packages/babel-plugin/src/utils/ast.tsx

```diff
@@ -450,7 +483,11 @@ export const resolveBindingNode = (
   referenceName: string,
   meta: Metadata
 ): PartialBindingWithMeta | undefined => {
-  const binding = meta.parentPath.scope.getBinding(referenceName);
+  // Check binding in own scope first so that manually created scopes can be
+  // evaluated first then parent scopes or scopes coming from different module.
+  const binding =
+    meta.ownPath?.scope.getOwnBinding(referenceName) ||
```

Comment:
when we say manually created scopes what are we talking about? us adding new variables into the scope?

are we missing the part where we add the binding to the scope? could we add it at a previous step?

Reply:
So the flow is:

- We pick the parameters
- Then evaluate the args by traversing
- Loop through params, and map the args with params
- Pull the params and create an IIFE around the calling function `CallExpression` which creates a new scope around calling function and isolates everything (We add bindings at this step)
- Add that IIFE node path to `ownPath`, and check only for own binding so that only isolated params are evaluated
- If at any step, `ownPath` is not set (module traversal case or any other case), it will pick things from `parentPath`

Since we are recursively traversing, its difficult to add bindings in `resolveBindingNode` because thats done under `traverseCallExpression` and `ownPath` is only variable which guarantees that recursively traversed things evaluates properly. 

Hope this makes sense 😅